compiler/il/OMRBlock.cpp

```diff
@@ -1472,11 +1472,35 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
             // If we can not find a register, store it into temp slot.
             if (!iter->second.second)
                {
-               iter->second.second = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectLoad(value->getDataType()), 0, ref);
+               TR::DataType valDataType = value->getDataType();
+               TR::Node *convertedValueNode  = value;
+               // In case for a given node, if the load/store (Minimum 4 bytes for alignment) size in memory is not same as node's data type,
+               // We need to convert the node to the type which can be stored into memory. Also we need to make sure that replacement node is a
+               // node that is converted back to original type of the node.
+               if (comp->fe()->dataTypeForLoadOrStore(valDataType) != valDataType)
+                  {
+                  TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(valDataType, comp->fe()->dataTypeForLoadOrStore(valDataType), false);
+                  convertedValueNode = TR::Node::create(convertOpCode, 1, value);
+                  }
+               TR::SymbolReference *storedValueSymRef = createSymRefForNode(comp, methodSymbol, convertedValueNode, self()->getExit());
+               self()->getExit()->insertBefore(TR::TreeTop::create(comp,
+                                                                     TR::Node::createWithSymRef(value,
+                                                                        comp->il.opCodeForDirectStore(convertedValueNode->getDataType()),
+                                                                        1,
+                                                                        convertedValueNode,
+                                                                        storedValueSymRef)));
+               TR::Node *replacementNode = TR::Node::createLoad(value, storedValueSymRef);
+               if (replacementNode->getDataType() != valDataType)
+                  {
+                  TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(replacementNode->getDataType(), valDataType, false);
+                  replacementNode = TR::Node::create(convertOpCode, 1, replacementNode);
+                  }
+               iter->second.second = replacementNode;
                }
             }
          TR::Node *replacement = iter->second.second;
-         // Now we know where the value will be stored handle the two cases in step 3 based on the opcode type
+         // If the replacement is regLoad node, attach it to GlRegDep node of the entry node of the new block and
+         // exit node of the original block.
```

Comment:
There is no reason to think we have to worry about conversions reaching this point? will it be correct if they do?

Reply:
I do not think so conversion will reach at this point. Previously, I was creating store to temp slot here which I changed with this commit so updated the comment. I do not think so we would need conversion in case we found the register to store the node.

Reply:
Ok I think if you do the refactor mentioned in the third comment I made on 1504 then I think this will be addressed - I'll resolve this once that update is done. Thanks!